architecture/2025-04-aci-confidential-containers.md

```diff
@@ -0,0 +1,388 @@
+  1. In the context of confidential containers, the CCE policy will be generated and set by Radius before the container group profile gets deployed
+2. Create an NGroups object 
+3. Create an ACI instance 
+
+In addition, we will need to support an `aci` property and `containergroupprofile` property under `runtimes`.
+
+### Detailed Design
+
+<!--
+This section should be detailed and thorough enough that another developer
+could implement your design and provide enough detail to get a high confidence
+estimate of the cost to implement the feature but isn’t as detailed as the 
+code. Be sure to also consider testability in your design.
+
+For each change, give each "change" in the proposal its own section and
+describe it in enough detail that someone else could implement it. Cover
+ALL of the important decisions like names. Your goal is to get an agreement
+to proceed with coding and PRs.
+
+If there are alternatives you are considering please include that in the open
+questions section. If the product has a layered architecture, it's good to
+align these sections with the product's layers. This will help readers use
+their current understanding to understand your ideas.
+
+Discuss the rationale behind architectural choices and alternative options 
+considered during the design process.
+-->
+
+#### Bicep changes  
+The Bicep schema will be updated to include a field for the `containergroupprofile` data and properties. These changes will be automatically generated through the TypeSpec schema changes. 
+
+#### Core RP changes
+
+**ACI Renderer**  
+The ACI renderer will need to be updated to support punchthrough values on the container group profile. Radius sets up a container group profile by default. If punchthrough values are provided by the user, the default values will be updated to use the user-provided values. This will be done through a merge process between the user inputs and the ContainerGroupProfile object provided by the ACI SDK. 
+```
+// ContainerGroupProfile - container group profile object
+type ContainerGroupProfile struct {
+	// The resource location.
+	Location *string `json:"location,omitempty"`
+
+	// The container group profile properties
+	Properties *ContainerGroupProfileProperties `json:"properties,omitempty"`
+
+	// The resource tags.
+	Tags map[string]*string `json:"tags,omitempty"`
+
+	// The zones for the container group.
+	Zones []*string `json:"zones,omitempty"`
+
+	// READ-ONLY; The resource id.
+	ID *string `json:"id,omitempty" azure:"ro"`
+
+	// READ-ONLY; The resource name.
+	Name *string `json:"name,omitempty" azure:"ro"`
+
+	// READ-ONLY; Metadata pertaining to creation and last modification of the resource.
+	SystemData *SystemData `json:"systemData,omitempty" azure:"ro"`
+
+	// READ-ONLY; The resource type.
+	Type *string `json:"type,omitempty" azure:"ro"`
+}
+```
+
+**Azure CG Profile handler**  
```

Comment:
Can any of these operations be done on the client side?

Reply:
We discussed requiring the user to define a container group profile and passing that to Radius, but ultimately decided against it to ensure that Radius handles the full setup for ACI